contracts/dev/DevAdmin.sol

```diff
@@ -182,4 +203,195 @@ contract DevAdmin is AccessControlInternal {
             ms.nodeIdToManufacturerName[tokenId] = name;
         }
     }
+
+    /**
+     * @notice Admin function to burn a list of vehicles and reset all its attributes
+     * @dev It reverts if any vehicle is paired
+     * @dev Caller must have the admin role
+     * @dev This contract has the BURNER_ROLE in the VehicleId
+     * @param tokenIds List of vehicle node ids
+     */
+    function adminBurnVehicles(uint256[] calldata tokenIds)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        address sdIdProxyAddress = SyntheticDeviceStorage
+            .getStorage()
+            .idProxyAddress;
+
+        uint256 tokenId;
+        address owner;
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            tokenId = tokenIds[i];
+
+            if (!INFT(vehicleIdProxyAddress).exists(tokenId))
+                revert InvalidNode(vehicleIdProxyAddress, tokenId);
+            if (ms.links[vehicleIdProxyAddress][tokenId] != 0)
```

Comment:
The names have gotten so confusing—can we make note that this is checking for aftermarket?